### PR TITLE
[AAE-15741] ADF CardView component - headers for multivalued strings, integers and floats

### PR DESCRIPTION
**Pull request by schattopadhyay1986**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Header for multivalued strings, floats and integers are not displayed.

**What is the new behaviour?**

https://github.com/Alfresco/alfresco-ng2-components/assets/121290499/38544670-c708-4216-b722-6137b6bf515b

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by wideLandscape — COMMENTED** [review]
looks good to me. Left 2 minor comments

File: lib/core/src/lib/card-view/components/card-view/card-view.component.html
Comment (schattopadhyay1986) [reply lib/core/src/lib/card-view/components/card-view/card-view.component.html:13]: > Are Chips used for/with "single" value property too? If not, maybe `displayLabelForChips` should be enough informative

Chips are used only for multivalued property. Will change name as 'displayLabelForChips'

File: lib/core/src/lib/card-view/components/card-view/card-view.component.spec.ts
Comment (chndn004) [reply lib/core/src/lib/card-view/components/card-view/card-view.component.spec.ts:36]: The purpose of beforeEach(waitForAsync(() => {...})) in the test suite is to handle asynchronous operations that might be involved in the setup phase of the test. Since the compileComponents() method is asynchronous, the waitForAsync function ensures that the test component and its dependencies are fully compiled and ready for testing before proceeding with the actual test cases. I have removed compileComponents as well as waitForAsync

Comment (chndn004) [reply lib/core/src/lib/card-view/components/card-view/card-view.component.spec.ts]: As discussed, used async/await instead of waitForAsync, and leave compileComponents

File: demo-shell/src/app/components/card-view/card-view.component.html
Comment (wideLandscape) [inline demo-shell/src/app/components/card-view/card-view.component.html:10]: I would change `showLabelForMultiValuedChip` to `showLabelForChips` too, to avoid additional mental steps

File: demo-shell/src/app/components/card-view/card-view.component.html
Comment (wideLandscape) [inline demo-shell/src/app/components/card-view/card-view.component.html:46]: same `toggleLabelForChips()`

File: lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.html
Comment (wideLandscape) [inline lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.html:57]: same

File: lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.stories.ts
Comment (wideLandscape) [inline lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.stories.ts:158]: `chipsWithLabelCardTextItem` or `displayLabelForChipsCardTextItem` ?

File: lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.ts
Comment (wideLandscape) [inline lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.ts:236]: `showLabelForChips` ?

**Comment by schattopadhyay1986** [comment]
@chndn004 please rename showLabelForMultiValuedChip() -> showLabelForChips() to make the similarity in card-view-textitem.component (both ts and html)

File: lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.ts
Comment (chndn004) [reply lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.ts]: Renamed

File: lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.html
Comment (chndn004) [reply lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.html]: Renamed

File: demo-shell/src/app/components/card-view/card-view.component.html
Comment (chndn004) [reply demo-shell/src/app/components/card-view/card-view.component.html]: Renamed

File: demo-shell/src/app/components/card-view/card-view.component.html
Comment (chndn004) [reply demo-shell/src/app/components/card-view/card-view.component.html]: Renamed

File: lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.stories.ts
Comment (chndn004) [reply lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.stories.ts]: Renamed as displayLabelForChipsCardTextItem

File: demo-shell/src/app/components/card-view/card-view.component.html
Comment (wideLandscape) [inline demo-shell/src/app/components/card-view/card-view.component.html:48]: I missed that... `Show label for chips property`?

Comment (chndn004) [reply demo-shell/src/app/components/card-view/card-view.component.html]: Renamed

**Review by wideLandscape — APPROVED** [review]
looks good to me! approving, but wait for some owner approval too 😄

File: lib/core/src/lib/card-view/components/card-view/card-view.component.ts
Comment (popovicsandras) [inline lib/core/src/lib/card-view/components/card-view/card-view.component.ts:63]: I'm not sure I understand, but how it is backward compatible this way? Didn't we say that we would keep the old behaviour as default?

File: lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.ts
Comment (popovicsandras) [inline lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.ts:62]: I'm not sure I understand, but how it is backward compatible this way? Didn't we say that we would keep the old behaviour as default?

File: lib/core/src/lib/card-view/components/card-view-item-dispatcher/card-view-item-dispatcher.component.ts
Comment (popovicsandras) [inline lib/core/src/lib/card-view/components/card-view-item-dispatcher/card-view-item-dispatcher.component.ts:61]: I'm not sure I understand, but how it is backward compatible this way? Didn't we say that we would keep the old behaviour as default?

**Review by popovicsandras — CHANGES_REQUESTED** [review]

File: lib/core/src/lib/card-view/components/card-view/card-view.component.ts
Comment (schattopadhyay1986) [reply lib/core/src/lib/card-view/components/card-view/card-view.component.ts]: Done.

File: lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.ts
Comment (schattopadhyay1986) [reply lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.ts]: Done.

File: lib/core/src/lib/card-view/components/card-view-item-dispatcher/card-view-item-dispatcher.component.ts
Comment (schattopadhyay1986) [reply lib/core/src/lib/card-view/components/card-view-item-dispatcher/card-view-item-dispatcher.component.ts]: Done.

**Review by popovicsandras — APPROVED** [review]

**Pull request merged**
